nixos/modules/programs/sway.nix

```diff
@@ -4,82 +4,30 @@ with lib;
 
 let
   cfg = config.programs.sway;
-  swayPackage = pkgs.sway;
-
-  swayWrapped = pkgs.writeShellScriptBin "sway" ''
-    if [[ "$#" -ge 1 ]]; then
-      exec sway-setcap "$@"
-    else
-      ${cfg.extraSessionCommands}
-      exec ${pkgs.dbus.dbus-launch} --exit-with-session sway-setcap
-    fi
-  '';
-  swayJoined = pkgs.symlinkJoin {
-    name = "sway-joined";
-    paths = [ swayWrapped swayPackage ];
-  };
+  swayPackage = cfg.package;
 in {
   options.programs.sway = {
     enable = mkEnableOption ''
-      the tiling Wayland compositor Sway. After adding yourself to the "sway"
-      group you can manually launch Sway by executing "sway" from a terminal.
-      If you call "sway" with any parameters the extraSessionCommands won't be
-      executed and Sway won't be launched with dbus-launch'';
-
-    extraSessionCommands = mkOption {
-      type = types.lines;
-      default = "";
-      example = ''
-        # Define a keymap (US QWERTY is the default)
-        export XKB_DEFAULT_LAYOUT=de,us
-        export XKB_DEFAULT_VARIANT=nodeadkeys
-        export XKB_DEFAULT_OPTIONS=grp:alt_shift_toggle,caps:escape
-        # Change the Keyboard repeat delay and rate
-        export WLC_REPEAT_DELAY=660
-        export WLC_REPEAT_RATE=25
-      '';
-      description = ''
-        Shell commands executed just before Sway is started.
-      '';
-    };
+      the tiling Wayland compositor Sway.
+    '';
 
-    extraPackages = mkOption {
-      type = with types; listOf package;
-      default = with pkgs; [
-        i3status xwayland rxvt_unicode dmenu
-      ];
-      defaultText = literalExample ''
-        with pkgs; [ i3status xwayland rxvt_unicode dmenu ];
-      '';
-      example = literalExample ''
-        with pkgs; [
-          i3lock light termite
-        ]
-      '';
+    package = mkOption {
+      type = types.package;
+      default = pkgs.sway-beta;
```

Comment:
I think the default should still be `sway` until 1.0.0 is officially released.

Reply:
I think that makes sense. Should I do a "split" module sort of thing where I keep the setcap wrapper unless `programs.sway.beta = true` or something?

Reply:
You could consider duplicating the module so we (temporarily) have a `sway` and a `sway-beta` module like we do for the package.  Once 1.0.0 is released, the `sway-beta` module and package will replace the previous `sway`. Maybe that's easier to maintain.

Reply:
@xeji seems simpler, yes, I'd be fine with it. Are we worried about introducing sway-beta and then removing it weeks later, from a deprecation, backward-compat standpoint? Maybe we don't care since none of this is likely to slip into 18.09 in this form?

Reply:
I wouldn't care at this point. It's called beta for a reason, so users should expect it to break. 
`sway-beta` shouldn't be backported to 18.09 anyway. But it might eventually end up in 19.03 unless sway 1.0.0 is released first... 
@primeos what do you think?

Reply:
I agree with @xeji, introducing a `sway-beta` module until the final release of Sway 1.0 sounds like the best idea. We could also use the description to mention that this module is only temporary.